src/test/integration/TeleportJoin.t.sol

```diff
@@ -0,0 +1,140 @@
+
+    bytes32 constant internal ILK = "TELEPORT-ETHEREUM-MASTER-1";
+    bytes32 constant internal MASTER_DOMAIN = "ETHEREUM-MASTER-1";
+    bytes32 constant internal SLAVE_DOMAIN = "L2NETWORK-SLAVE-1";
+
+    ChainlogLike internal chainlog = ChainlogLike(0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F);
+    EndLike internal end = EndLike(chainlog.getAddress("MCD_END"));
+    CureLike internal cure = CureLike(chainlog.getAddress("MCD_CURE"));
+    VatLike internal vat = VatLike(chainlog.getAddress("MCD_VAT"));
+    address internal vow = chainlog.getAddress("MCD_VOW");
+
+    TeleportJoin internal teleportJoin;
+
+    uint256 internal constant RAD = 10**45;
+    uint256 internal constant RAY = 10**27;
+    uint256 internal constant TTL = 8 days;
+
+    function getAuthFor(address auth) internal {
+        hevm.store(
+            auth,
+            keccak256(abi.encode(address(this), 0)),
+            bytes32(uint256(1))
+        );
+        assertEq(AuthLike(auth).wards(address(this)), 1);
+    }
+
+    function setUp() public {
+        // setup teleportJoin
+        teleportJoin = new TeleportJoin(address(vat), chainlog.getAddress("MCD_JOIN_DAI"), ILK, MASTER_DOMAIN);
+        teleportJoin.file(bytes32("vow"), vow);
+        teleportJoin.file("line", SLAVE_DOMAIN, 1_000_000 ether);
+        teleportJoin.file("fees", SLAVE_DOMAIN, address(new TeleportConstantFee(0, TTL)));
+
+        // setup ILK in vat
+        getAuthFor(address(vat));
+        vat.rely(address(teleportJoin));
+        vat.init(ILK);
+        vat.file(ILK, bytes32("spot"), RAY);
+        vat.file(ILK, bytes32("line"), 10000000000 * RAD);
+
+        // setup cure
+        getAuthFor(address(cure));
+        cure.lift(address(teleportJoin));
+    }
+
+    function testEmergencyShutdown() public {
+        // perform teleport
+
+        uint256 debtBeforeTeleport = vat.debt(); 
+        uint256 teleportAmount = 250_000 ether;
+        TeleportGUID memory guid = TeleportGUID({
+            sourceDomain: SLAVE_DOMAIN,
+            targetDomain: MASTER_DOMAIN,
+            receiver: addressToBytes32(address(123)),
+            operator: addressToBytes32(address(654)),
+            amount: uint128(teleportAmount),
+            nonce: 5,
+            timestamp: uint48(block.timestamp)
+        });
+
+        teleportJoin.requestMint(guid, 0, 0);
+
+        assertEq(vat.debt(), debtBeforeTeleport + teleportAmount * RAY);
+
+        // cage the end
+
+        getAuthFor(address(end));
+
+        end.cage();
+
+        // load the cure 
+
+        cure.load(address(teleportJoin));
+
+        assertEq(cure.tell(), teleportAmount * RAY);
+
+        // thaw the end
+
+        uint256 vatDebt = vat.debt();
+        hevm.warp(block.timestamp + end.wait());
+        hevm.store(address(vat),keccak256(abi.encode(vow, 5)), bytes32(0)); // emulate clearing of vow dai
+        assertEq(vat.dai(vow), 0);
+
+        end.thaw();
+
+        assertEq(end.debt(), vatDebt - teleportAmount * RAY);
```

Comment:
Maybe we just want to finish the cycle and show `settle` will leave the `teleportAmount` locked in the `join` address. Not sure if really useful though.

Reply:
I've added an attempt to `settle` the debt after the end is`cage`d to the test. Let me know if that's not what you had in mind.

Reply:
Yes this looks good to me! thanks